### PR TITLE
NN-3288: Stop duplicate inmates on desc

### PR DESCRIPTION
**Pull request by sp-watson**
Previously it returned one offender booking for each `OFFENDER_IMPRISON_STATUSES` entry for that booking, of which there can be any number.

Looking at the code, it appears that this is a problem elsewhere, e.g. any call to `GlobalSearchService.executeQuery()`, but that needs to be addressed in a separate ticket

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/repository/sql/InmateRepositorySql.kt [hunk 1]
```diff
@@ -185,11 +185,14 @@ enum class InmateRepositorySql(val sql: String) {
...
         FROM OFFENDER_BOOKINGS OB
-        INNER JOIN OFFENDERS O ON OB.OFFENDER_ID = O.OFFENDER_ID
-                LEFT JOIN AGENCY_INTERNAL_LOCATIONS AIL ON OB.LIVING_UNIT_ID = AIL.INTERNAL_LOCATION_ID
-                LEFT JOIN OFFENDER_IMPRISON_STATUSES OIS ON OIS.OFFENDER_BOOK_ID = OB.OFFENDER_BOOK_ID AND OIS.LATEST_STATUS = 'Y'
-        LEFT JOIN IMPRISONMENT_STATUSES IST ON IST.IMPRISONMENT_STATUS = OIS.IMPRISONMENT_STATUS
-                WHERE OB.ACTIVE_FLAG = 'Y' AND OB.BOOKING_SEQ = 1
+            INNER JOIN OFFENDERS O ON OB.OFFENDER_ID = O.OFFENDER_ID
+            LEFT JOIN AGENCY_INTERNAL_LOCATIONS AIL ON OB.LIVING_UNIT_ID = AIL.INTERNAL_LOCATION_ID
+            LEFT JOIN (
+                SELECT OIST.* FROM OFFENDER_IMPRISON_STATUSES OIST WHERE OIST.IMPRISON_STATUS_SEQ IN
```
Comment (JamieDraperUK) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/sql/InmateRepositorySql.kt:191]: Would it be better for performance to select specific columns? Not sure if it'll make a massive difference but you never know.

File: src/main/java/uk/gov/justice/hmpps/prison/repository/sql/InmateRepositorySql.kt [hunk 2]
```diff
@@ -185,11 +185,14 @@ enum class InmateRepositorySql(val sql: String) {
...
-        INNER JOIN OFFENDERS O ON OB.OFFENDER_ID = O.OFFENDER_ID
-                LEFT JOIN AGENCY_INTERNAL_LOCATIONS AIL ON OB.LIVING_UNIT_ID = AIL.INTERNAL_LOCATION_ID
-                LEFT JOIN OFFENDER_IMPRISON_STATUSES OIS ON OIS.OFFENDER_BOOK_ID = OB.OFFENDER_BOOK_ID AND OIS.LATEST_STATUS = 'Y'
-        LEFT JOIN IMPRISONMENT_STATUSES IST ON IST.IMPRISONMENT_STATUS = OIS.IMPRISONMENT_STATUS
-                WHERE OB.ACTIVE_FLAG = 'Y' AND OB.BOOKING_SEQ = 1
+            INNER JOIN OFFENDERS O ON OB.OFFENDER_ID = O.OFFENDER_ID
+            LEFT JOIN AGENCY_INTERNAL_LOCATIONS AIL ON OB.LIVING_UNIT_ID = AIL.INTERNAL_LOCATION_ID
+            LEFT JOIN (
+                SELECT OIST.* FROM OFFENDER_IMPRISON_STATUSES OIST WHERE OIST.IMPRISON_STATUS_SEQ IN 
+                (SELECT MAX(IMPRISON_STATUS_SEQ) FROM OFFENDER_IMPRISON_STATUSES WHERE LATEST_STATUS = 'Y' AND OFFENDER_BOOK_ID = OIST.OFFENDER_BOOK_ID)
```
Comment (JamieDraperUK) [inline src/main/java/uk/gov/justice/hmpps/prison/repository/sql/InmateRepositorySql.kt:192]: Is this at risk of reaching the maximum values passed into an IN clause? I think it's a 1000.

**Review by JamieDraperUK — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/hmpps/prison/repository/sql/InmateRepositorySql.kt (hunk 1 above)
Comment (sp-watson) [reply src/main/java/uk/gov/justice/hmpps/prison/repository/sql/InmateRepositorySql.kt:191]: yeah, suppose so

**Review by JamieDraperUK — APPROVED** [review]
LGTM

**Pull request merged**